project/npda/general_functions/csv_upload.py

```diff
@@ -782,311 +782,293 @@ def validate_date(
             site_errors,
         )
 
-    # private method - saves the csv row in the model as a record
-    def save_row(row, timestamp, cohort_id):
-        """
-        Save each row as a record
-        First validate the values in the row, then create a Patient instance - if contains invalid items, set is_valid to False, with the error messages
-        Then use the Patient instance to create a Visit instance, again testing values first and storing if is_valid and errors
-        """
-        (
-            patient_is_valid,
-            patient_errors,
-            visit_is_valid,
-            visit_errors,
-            site_is_valid,
-            site_errors,
-        ) = validate_row(row)
+    # private method - saves the csv rows for a given patient
+    def save_rows(rows, timestamp, new_cohort):
+        rows["validation_result"] = rows.apply(validate_row, axis = 1)
 
-        # save the site
-        try:
-            # save site
-            site, created = Site.objects.update_or_create(
-                date_leaving_service=(
-                    row["Date of leaving service"]
-                    if not pd.isnull(row["Date of leaving service"])
-                    else None
-                ),
-                reason_leaving_service=(
-                    row["Reason for leaving service"]
-                    if not pd.isnull(row["Reason for leaving service"])
-                    else None
-                ),
-                paediatric_diabetes_unit_pz_code=pdu_pz_code,
-                organisation_ods_code=organisation_ods_code,
-            )
-        except Exception as error:
-            raise Exception(f"Could not save site: {error}")
```

Comment:
These try/catches just tended to hide the stack trace during development which was very annoying so I've removed them. We may want to add them back to avoid an angry blank screen, but tbh custom error pages are probably the better way to go (https://github.com/rcpch/national-paediatric-diabetes-audit/issues/21) so that we are sure we will get full stack trace logging in the backend logs